### PR TITLE
MCP temperaray exception handling fix

### PR DESCRIPTION
**Pull request by CoronRing**
feat(rt_mcp): Improve MCP error handling, OAuth support, and documentation

Major improvements to MCP system reliability and developer experience:

🐛 Bug Fixes:
- Fixed critical bug where exceptions in background thread weren't propagated to main thread
- Added _setup_exception field to MCPServer to capture and re-raise exceptions with context
- Enhanced close() method to safely handle None values for _loop and _shutdown_event
- Fixed race condition where _ready_event was set even on failure

⚡ Error Handling Improvements:
- Added setup_timeout parameter (default 30s) to all connection methods
- Enhanced error messages with helpful troubleshooting context:
  * FileNotFoundError: Suggests checking PATH and command spelling
  * ConnectionError: Provides network/auth troubleshooting steps
  * TimeoutError: Includes command/URL in error message
- Added proper timeout handling to _ready_event.wait()

✅ Test Suite Additions (22 tests, 100% passing in <1s):
- test_error_handling.py: 7 tests for error propagation and cleanup
  * Invalid commands and exception handling
  * Error message context validation
  * Thread cleanup verification
  * Close() method edge cases
- test_oauth.py: 15 tests for OAuth integration
  * OAuth parameter handling
  * Token storage operations
  * Metadata creation and validation
  * Full integration tests
- All tests use mocks (no external network calls)
- Zero thread leaks after errors

🔧 Testing Infrastructure:
- Created test_mcp_debug.py for MCP diagnostics (8 tests)
- Created test_deepwiki_mcp.py for DeepWiki integration testing
- Created check_deepwiki_params.py for tool parameter inspection
- All test files use proper fixtures and mocking

📖 Additional Documentation:
- TEST_SUITE_ADDITION_SUMMARY.md: Comprehensive test suite overview
- Improved code readability with better variable names and structure

This commit resolves critical stability issues, adds professional OAuth support,
and significantly improves the developer experience with better error messages
and comprehensive documentation.

### CODE REVIEW

File: packages/railtracks/tests/unit_tests/rt_mcp/test_oauth.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+            auth=mock_oauth,
+        )
+        
+        assert config.url == "http://localhost:8000/mcp"
+        assert config.auth == mock_oauth
+        assert config.timeout.total_seconds() == 30
+
+    def test_oauth_params_validation(self):
+        """Test that OAuth parameters are properly validated."""
+        from railtracks.rt_mcp import MCPHttpParams
```
Comment (soulFood5632) [inline packages/railtracks/tests/unit_tests/rt_mcp/test_oauth.py]: local import that shouldn't be

File: packages/railtracks/src/railtracks/rt_mcp/mcp_tool.py [hunk 2]
```diff
@@ -5,20 +5,85 @@
...
-    The connection will remain open until the server is closed with `close()`.
+    Connect to an MCP server and return a server instance with available tools.
+
+    This is the primary entry point for using MCP servers in Railtracks.
+    The server will connect in the background, discover available tools,
+    and convert them to Railtracks Node classes.
+
+    The connection remains active until explicitly closed or the context exits.
+
+    Usage Examples:
```
Comment (Amir-R25) [inline packages/railtracks/src/railtracks/rt_mcp/mcp_tool.py:21]: I think it's best we remove Examples from docstrings

Comment (CoronRing) [reply packages/railtracks/src/railtracks/rt_mcp/mcp_tool.py:21]: I think it should be fine, the usage is not intuiative, easier to understand and use by leaving docs there. I think style << ability for ai to understand

Comment (CoronRing) [reply packages/railtracks/src/railtracks/rt_mcp/mcp_tool.py:21]: @Amir-R25 Do you mean this comment or soulfood one above? if the one above it should be resolved last week

File: packages/railtracks/tests/unit_tests/rt_mcp/test_oauth.py (hunk 1 above)
Comment (CoronRing) [reply packages/railtracks/tests/unit_tests/rt_mcp/test_oauth.py]: All removed and updated

File: packages/railtracks/src/railtracks/rt_mcp/mcp_tool.py (hunk 2 above)
Comment (Amir-R25) [reply packages/railtracks/src/railtracks/rt_mcp/mcp_tool.py:21]: The one above, perfect thanks!
